### PR TITLE
Add source tests for deleting user-defined & system wide repositories

### PR DESCRIPTION
**Pull request by atodorov**
Between the existing sanity tests and this PR the behavior documented in #315 is covered.

### CODE REVIEW

**Comment by atodorov** [comment]
@bcl can you take a look at this ? Should be OK to merge. Thanks.

**Comment by bcl** [comment]
Let's wait until PR #404 is merged, then rebase it and make sure it works as a unit test as well.

File: internal/client/source_test.go [hunk 1]
```diff
@@ -183,3 +187,92 @@ func TestGetSourceInfoV0(t *testing.T) {
...
+func TestDeleteSystemSourcesV0(t *testing.T) {
+	hostDistroName, err := distro.GetHostDistroName()
+	hostArch := common.CurrentArch()
+	require.NoError(t, err)
+
+	// note: this is an integration test and it expects osbuild-composer to be installed
+	// that's why we search for repositories configuration in system wide directories!
+	repoMap, err := rpmmd.LoadRepositories([]string{"/etc/osbuild-composer", "/usr/share/osbuild-composer"}, hostDistroName)
+	require.NoError(t, err, "rpmmd.LoadRepositories: %v", err)
+	require.GreaterOrEqual(t, len(repoMap[hostArch]), 1)
```
Comment (atodorov) [inline internal/client/source_test.go]: @bcl after the changes in #404 I'm not sure how these tests are expected to be executed and if repository definitions should be read from the system wide directories. Can you comment on this?

**Comment by atodorov** [comment]
Failures here don't seem related to the proposed changes.

File: internal/client/source_test.go (hunk 1 above)
Comment (atodorov) [reply internal/client/source_test.go]: After the rebase this now fails b/c I have only a "test-id" repository available and not the fedora and updates repositories, although I am trying to load them.

Comment (atodorov) [reply internal/client/source_test.go]: @bcl I am starting to think this better become an integration test. What do you think ?

Comment (atodorov) [reply internal/client/source_test.go]: Scratch that. I've managed to rework this test and now it tries to delete the repository which comes configured with the API server.

Comment (bcl) [reply internal/client/source_test.go]: I'm not seeing that commit, it's still trying to delete "test-system-repo" during the integration test.

Comment (atodorov) [reply internal/client/source_test.go]: In my previous comment I meant "test-system-repo" (before that I had a different implementation). 

The latest update tries several well known source names and skips them if they aren't available in the initial sources list. That makes the test fault  tolerant and you can execute it both as unit test and as an integration test. In integration test mode I get a failure which doesn't seem related to the current changes.

**Pull request merged**